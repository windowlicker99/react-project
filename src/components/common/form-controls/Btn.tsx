/* eslint-disable @typescript-eslint/no-empty-function */
import React, { FC, MouseEventHandler } from 'react';
import classNames from 'classnames';

interface IBtnProps {
  btnStatus: string;
  btnIcon?: string;
  btnText?: string;
  disabledBtn?: boolean;
  onClick?: MouseEventHandler;
  dataFor?: string;
}

export const Btn: FC<IBtnProps> = ({
  btnStatus = '',
  btnIcon = '',
  btnText = '',
  disabledBtn = false,
  onClick = () => {},
  dataFor = '',
}) => (
  <button
    type="button"
    disabled={disabledBtn}
    className={classNames('btn', btnStatus)}
    onClick={onClick}
    {...(dataFor && { 'data-for': dataFor, 'data-tip': true })}
  >
    <span className={`btn-icon ${btnIcon}`} />
    <span className="btn-text">{btnText}</span>
  </button>
);
