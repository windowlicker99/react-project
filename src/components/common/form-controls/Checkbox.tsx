import React, { ChangeEventHandler, FC } from 'react';

interface ICheckboxProps {
  name: string;
  checked?: boolean;
  label?: string;
  onChange: ChangeEventHandler<HTMLInputElement>;
  id?: string;
}

export const Checkbox: FC<ICheckboxProps> = ({ name, checked = false, label = '', onChange, id }) => (
  <div className="checkbox">
    <label className="checkbox-label" htmlFor={id || name}>
      <input
        name={name}
        className="checkbox-input"
        type="checkbox"
        checked={checked}
        id={id || name}
        onChange={onChange}
      />
      <span className="checkbox-mark" />
      <span className="checkbox-title">{label}</span>
    </label>
  </div>
);
