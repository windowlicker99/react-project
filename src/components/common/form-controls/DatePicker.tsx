import { getMonth, getYear } from 'date-fns';
import classNames from 'classnames';
import React, { FC } from 'react';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import { months } from '@constants/date';
import { rangeOfYears } from '@utils/formatDate';
import { convertListToOptions } from '@utils/converters';
import { IOption } from '@interfaces/interfaces';
import { getOptionIndex } from '@utils/getOptionIndex';
import { SelectComponent } from './SelectComponent';
import { ValidationErrorMessage } from './ValidationErrorMessage';

interface DatePickerProps {
  formLabel?: string;
  readOnly?: boolean;
  name?: string;
  value?: Date;
  error?: string;
  minDate?: string;
  onChange?: (name: string, value: Date) => void;
}

export const DatePickerComponent: FC<DatePickerProps> = ({
  formLabel = '',
  readOnly = false,
  name,
  value = null,
  error = '',
  minDate,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  onChange = () => {},
}) => {
  const yearsOptions = convertListToOptions(rangeOfYears(1900, 2100));
  const monthOptions = convertListToOptions(months);
  return (
    <div className={classNames('form-group', { error: Boolean(error) })}>
      <label htmlFor="exampleFormControlInput1" className="form-label">
        {formLabel}
      </label>
      <DatePicker
        placeholderText="Not Selected"
        autoComplete="off"
        readOnly={readOnly}
        dateFormat="dd/MM/yyyy"
        selected={value}
        name={name}
        minDate={minDate && new Date(minDate)}
        onChange={(date: Date) => onChange(name, date)}
        renderCustomHeader={({
          date,
          changeYear,
          changeMonth,
          decreaseMonth,
          increaseMonth,
          prevMonthButtonDisabled,
          nextMonthButtonDisabled,
        }) => (
          <div className="datepicker-control">
            <button
              className="datepicker-btn "
              type="button"
              onClick={decreaseMonth}
              disabled={prevMonthButtonDisabled}
            >
              <span className="icon-arrow-border-left" />
            </button>
            <SelectComponent
              defaultOption={getOptionIndex(yearsOptions, getYear(date))}
              options={yearsOptions}
              onChange={(_, { value }: IOption) => changeYear(Number(value))}
            />
            <SelectComponent
              options={monthOptions}
              defaultOption={getMonth(date)}
              onChange={(_, { value }: IOption) => changeMonth(months.indexOf(value))}
            />

            <button className="datepicker-btn" type="button" onClick={increaseMonth} disabled={nextMonthButtonDisabled}>
              <span className="icon-arrow-border-right" />
            </button>
          </div>
        )}
      />
      <ValidationErrorMessage error={error} />
    </div>
  );
};
