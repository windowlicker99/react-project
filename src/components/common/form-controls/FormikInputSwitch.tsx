import React, { FC, useState } from 'react';
import { FieldProps } from 'formik';
import { Input } from './Input';
import { SwitchFormik } from './SwitchFormik';

interface IFormikInputSwitch {
  label?: string;
  placeholder?: string;
  disabled?: boolean;
}

export const FormikInputSwitch: FC<IFormikInputSwitch & FieldProps> = ({
  field,
  label = '',
  placeholder = 'placeholder',
  disabled = false,
}) => {
  const [isSelected, setIsSelected] = useState(true);
  return (
    <div className="form-input-switch">
      <Input name="test" label={label} placeholder={placeholder} disabled={disabled} isSelected={isSelected} />
      <SwitchFormik field={field} setIsSelected={setIsSelected} />
    </div>
  );
};
