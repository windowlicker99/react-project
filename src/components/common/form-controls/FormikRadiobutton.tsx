import { FieldProps } from 'formik';
import React, { FC } from 'react';

interface IFormikRadiobuttonProps {
  label: string;
}

export const FormikRadiobutton: FC<IFormikRadiobuttonProps & Partial<FieldProps>> = ({
  label,
  field: { value, ...field },
}) => (
  <div className="radio-button">
    <label className="radio-label">
      <input type="radio" name="radio" className="radio-input" value={label} checked={value === label} {...field} />
      <span className="radio-mark" />

      <span className="radio-title">{label}</span>
    </label>
  </div>
);
