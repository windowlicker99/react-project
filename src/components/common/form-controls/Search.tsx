/* eslint-disable @typescript-eslint/no-empty-function */
import React, { FC, ChangeEvent } from 'react';

interface ISearchProps {
  formLabel?: string;
  formPlaceholder: string;
  searchText?: string;
  onSearch?: (arg0: string) => void;
}

export const Search: FC<ISearchProps> = ({ formLabel = '', formPlaceholder, searchText = '', onSearch = () => {} }) => {
  const onChange = (e: ChangeEvent<HTMLInputElement>) => onSearch(e.currentTarget.value);
  const onResetClick = () => onSearch('');

  return (
    <div className="form-group">
      <label htmlFor="exampleFormControlInput1" className="form-label search-label">
        {formLabel}
      </label>
      <div className="search-group">
        <div className="search-icon">
          <span className="icon-loupe" />
        </div>
        <input
          type="email"
          className="form-control search-input"
          id="exampleFormControlInput1"
          placeholder={formPlaceholder}
          value={searchText}
          onChange={onChange}
        />
        <button type="button" className="search-btn" onClick={onResetClick}>
          <span className="icon-cross" />
        </button>
      </div>
    </div>
  );
};
