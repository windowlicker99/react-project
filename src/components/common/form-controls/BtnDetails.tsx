import React, { FC } from 'react';
import { useHistory } from 'react-router';
import { Btn } from './Btn';

interface BtnDetailsProps {
  BtnDetailsText: string;
  path: string;
}

export const BtnDetails: FC<BtnDetailsProps> = ({ BtnDetailsText, path }) => {
  const history = useHistory();
  const onClick = () => {
    if (path) {
      history.push(path);
    }
  };
  return (
    <div role="button" tabIndex={0} className="btn-details" onClick={onClick} onKeyDown={onClick}>
      <span className="btn-detail-text">{BtnDetailsText}</span>
      <Btn btnStatus="btn-xs-primary" btnIcon="icon-arrow-border-right" />
    </div>
  );
};
