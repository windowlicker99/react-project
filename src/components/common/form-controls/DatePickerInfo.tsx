import React, { FC } from 'react';
import { DatePickerComponent } from './DatePicker';

interface IDatePickerInfo {
  label: string;
  datePickerLabel?: string;
  value: Date;
  name: string;
  error: string;
  readOnly: boolean;
  minDate?: string;
  onChange?: (name: string, value: Date) => void;
}

export const DatePickerInfo: FC<IDatePickerInfo> = ({ label, error, datePickerLabel, ...datePicker }) => (
  <div className="controller-info">
    <span className="controller-info-title">{label}</span>
    <DatePickerComponent formLabel={datePickerLabel} error={error} {...datePicker} />
  </div>
);
