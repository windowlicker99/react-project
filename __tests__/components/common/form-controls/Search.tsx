import React from 'react';
import { render } from '@testing-library/react';
import { Search } from '@components/common/form-controls/Search';

describe('Search tests', () => {
  it('Should render', () => {
    const { getByLabelText } = render(
      <Search formLabel="label" formPlaceholder="placeholder" />,
    );
    const text = getByLabelText('label');
    expect(text).toBeTruthy();
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(
      <Search formLabel="label" formPlaceholder="placeholder" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
