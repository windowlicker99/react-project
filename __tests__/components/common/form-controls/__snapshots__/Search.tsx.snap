// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class="form-label search-label"
      for="exampleFormControlInput1"
    >
      label
    </label>
    <div
      class="search-group"
    >
      <div
        class="search-icon"
      >
        <span
          class="icon-loupe"
        />
      </div>
      <input
        class="form-control search-input"
        id="exampleFormControlInput1"
        placeholder="placeholder"
        type="email"
        value=""
      />
      <button
        class="search-btn"
        type="button"
      >
        <span
          class="icon-cross"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
